src/components/base-message.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+class BaseMessage {
+
+  constructor(teamId, channelId, userId, content, originalMessage) {
+    this.teamId = teamId;
+    this.channelId = channelId;
+    this.userId = userId;
+    this.content = content;
+    this.originalMessage = originalMessage;
+
+  }
+
+  static createFromSlack(message) {
+    return new BaseMessage(message.team_id, message.channel, message.user, message.text, message);
+  }
+
+  static createFromDiscord(message) {
+    return new BaseMessage(message.guild.id, message.channel.id, message.author.id, message.content, message);
```

Comment:
This throws an error on trying to send a DM to the Bot, because DM messages do not have any guild associated.

Suggesting to check if message.guild is defined, and if not, then just pass "null" as the value.

Also suggest to pass "null" in "createFromSlack" if "message.team_id" is undefined.